src/overlays/actors/ovl_Bg_Mizu_Water/z_bg_mizu_water.c

```diff
@@ -21,19 +43,305 @@ const ActorInit Bg_Mizu_Water_InitVars = {
     (ActorFunc)BgMizuWater_Update,
     (ActorFunc)BgMizuWater_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mizu_Water/func_8089F8B0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mizu_Water/func_8089F9D0.s")
+static u32 unk_808A0550 = 0;
+static f32 unk_808A0554 = 110.0f;
+
+static u32 waterBoxIDs[] = { 0x00000002, 0x00000003, 0x00000005, 0x00000007,
+                             0x0000000C, 0x00000014, 0x00000015, 0x00000016 };
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F(scale, 1, ICHAIN_STOP),
+};
+
+u32 BgMizuWater_getWaterLevelActionIndex(s16 switchID, GlobalContext* globalCtx) {
```

Comment:
The naming scheme is `SystemName_FuncName` so this lacks capitalization (same for `BgMizuWater_setWaterBoxesHeights`)

Also why `Action`? the result indexes into `waterLevels` so it makes more sense to just call it `waterLevelIndex`?

Reply:
Yea, I'm not too happy with that name either. My idea with calling it `Action` was that it returns 0 if the water level doesn't have to be changed and a water level index otherwise, which is used for changing the water level (it also sets the switches based on `bREG(0)` and `bREG(1)`). But yeah, probably a bad name anyway.